Sources/SceytChatUIKit/Providers/ChannelMessageMarkerProvider.swift

```diff
@@ -113,18 +113,70 @@ open class ChannelMessageMarkerProvider: Provider {
             return
         }
         logger.debug("[MARKER CK] send \(ids.count)")
-        switch markerName {
-        case DefaultMarker.received:
+        switch DefaultMarker(rawValue: markerName) {
+        case .received:
             channelOperator.markMessagesAsReceived(ids: ids.map { NSNumber(value: $0)}, completion: completed(_:error:))
-        case DefaultMarker.displayed:
+        case .displayed:
             channelOperator.markMessagesAsDisplayed(ids: ids.map { NSNumber(value: $0)}, completion: completed(_:error:))
         default:
             channelOperator.markMessages(markerName: markerName, ids: ids.map { NSNumber(value: $0)}, completion: completed(_:error:))
         }
     }
 }
 
-enum DefaultMarker {
-    static let displayed = "displayed"
-    static let received = "received"
+extension ChannelMessageMarkerProvider {
+    
+    open func loadMarkers(_ query: MessageMarkerListQuery, completion: ((Error?) -> Void)? = nil) {
+        guard chatClient.connectionState == .connected else { return }
+        guard query.hasNext, !query.loading else { return }
+        
+        query.loadNext { [weak self] query, markers, error in
+            guard let self else { return }
+            
+            if let error {
+                logger.errorIfNotNil(error, "")
+            } else {
+                if let markers {
+                    store(markers: markers, for: query.messageId, completion: completion)
+                }
+            }
+        }
+    }
+    
+    open func store(markers: [Marker],
+                    for messageId: MessageId,
+                    completion: ((Error?) -> Void)? = nil) {
+        guard !markers.isEmpty else {
+            completion?(nil)
+            return
+        }
+        database.write {
+            $0.createOrUpdate(markers: markers, messageId: messageId)
+            
+        } completion: { error in
+            logger.debug(error?.localizedDescription ?? "")
+            completion?(error)
+        }
+    }
+}
+
+enum DefaultMarker: String, Comparable {
+    case displayed
+    case received
+    case played
```

Comment:
@arthursct jan did you check is it possible create .custom(string) marker
for example, if someone wants to add 4th marker, how could it be